packages/flutter/lib/src/material/page.dart

```diff
@@ -85,59 +96,37 @@ class MaterialPageRoute<T> extends PageRoute<T> {
 
   @override
   bool canTransitionFrom(TransitionRoute<dynamic> nextRoute) {
-    return nextRoute is MaterialPageRoute<dynamic>;
+    return nextRoute is MaterialPageRoute<dynamic> || nextRoute is CupertinoPageRoute<dynamic>;
```

Comment:
It's slightly semantically different I guess. _cupertinoPageRoute doesn't have to exist yet for the current one to be able to transition from and to the next one.